lib/src/runner/node/platform.dart

```diff
@@ -0,0 +1,194 @@
+  /// source map for the compiled suite.
+  Future<Pair<Process, StackTraceMapper>> _spawnProcess(
+      String path, SuiteConfiguration suiteConfig) async {
+    var dir = new Directory(_compiledDir).createTempSync('test_').path;
+    var jsPath = p.join(dir, p.basename(path) + ".node_test.dart.js");
+
+    if (_config.pubServeUrl == null) {
+      await _compilers.compile(
+          '''
+        import "package:test/src/bootstrap/node.dart";
+
+        import "${p.toUri(p.absolute(path))}" as test;
+
+        void main() {
+          internalBootstrapNodeTest(() => test.main);
+        }
+      ''',
+          jsPath,
+          suiteConfig);
+
+      // Add the Node.js preamble to ensure that the dart2js output is
+      // compatible. Use the minified version so the source map remains valid.
+      var jsFile = new File(jsPath);
+      await jsFile.writeAsString(
+          preamble.getPreamble(minified: true) + await jsFile.readAsString());
+
+      StackTraceMapper mapper;
+      if (!suiteConfig.jsTrace) {
+        var mapPath = jsPath + '.map';
+        mapper = new StackTraceMapper(await new File(mapPath).readAsString(),
+            mapUrl: p.toUri(mapPath),
+            packageResolver: await PackageResolver.current.asSync,
+            sdkRoot: p.toUri(sdkDir));
+      }
+
+      return new Pair(await Process.start(_executable, [jsPath]), mapper);
+    }
+
+    var url = _config.pubServeUrl.resolveUri(
+        p.toUri(p.relative(path, from: 'test') + '.node_test.dart.js'));
+
+    var js = await _get(url, path);
+    await new File(jsPath)
+        .writeAsString(preamble.getPreamble(minified: true) + js);
+
+    StackTraceMapper mapper;
+    if (!suiteConfig.jsTrace) {
+      var mapUrl = url.replace(path: url.path + '.map');
+      mapper = new StackTraceMapper(await _get(mapUrl, path),
+          mapUrl: mapUrl,
+          packageResolver: new SyncPackageResolver.root('packages'),
+          sdkRoot: p.toUri('packages/\$sdk'));
+    }
+
+    return new Pair(await Process.start(_executable, [jsPath]), mapper);
+  }
+
+  /// Runs an HTTP GET on [url].
+  ///
+  /// If this fails, throws a [LoadException] for [suitePath].
+  Future<String> _get(Uri url, String suitePath) async {
+    try {
+      var response = await (await _http.getUrl(url)).close();
+
+      if (response.statusCode != 200) {
+        // We don't care about the response body, but we have to drain it or
+        // else the process can't exit.
+        response.listen(null);
+
+        throw new LoadException(
+            suitePath,
+            "Error getting $url: ${response.statusCode} "
+            "${response.reasonPhrase}\n"
+            'Make sure "pub serve" is serving the test/ directory.');
+      }
+
+      return await UTF8.decodeStream(response);
+    } on IOException catch (error) {
+      var message = getErrorMessage(error);
+      if (error is SocketException) {
+        message = "${error.osError.message} "
```

Comment:
okay I'm going to stop posting about this quote thing, but check out other places in these files that it's happening.